hs_core/views/__init__.py

```diff
@@ -484,6 +485,11 @@ def file_download_url_mapper(request, shortkey):
     resource, _, _ = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE)
     istorage = resource.get_irods_storage()
     irods_file_path = '/'.join(request.path.split('/')[2:-1])
+    listing = istorage.listdir('/'.join(irods_file_path.split('/')[0:-1]))
+    if irods_file_path.split('/')[-1] in listing[0]:
+        # it's a folder
+        file_download_url = istorage.url(os.path.join('zips', irods_file_path))
+        return HttpResponseRedirect(file_download_url)
```

Comment:
@sblack-usu I don't think you need to add this to this function since this function is one way street to map url for each file encoded in resourcemap.xml to a valid file download url. Temporary folder zip file will never become part of resourcemap.xml, so no need for this code snippet here.

Reply:
This method is used by hs_core/resourcemap_urls.py for downloading a file.  I updated this method, so it could recognize folders and correctly route to the download zips method.  I had not considered that this method is used by the resourcemap.xml.  Are you sure it's used to create the resourcemp.xml?  It's return a response rather than a url so intuitively I wouldn't think it'd be used for setting up urls in the resourcemap.

Reply:
@sblack-usu I meant this view function is added to map the URIs encoded in resourcemap.xml (i.e., in format of /resource/data/contents/filename) to the internal download url so that typing in URIs encoded in resourcemap.xml would result in a file download. The zip file will not become part of resourcemap.xml, so thought the internal download URL can be used without need for external URL mapping. Am I missing something?

Reply:
right, so I'm just introducing the ability for the url mapping for /resource with a folder (i.e. /resource/data/contents/folder).  The method is expanded to check whether the last path parameter is a file or folder and if it's a folder, it maps it to the url to download the folder.  For clarification, the url that starts the zipping process for a folder and redirects to the temporary zipped folder.

Is there a different external url that you have in mind?

Reply:
Just one thing to confirm: is folder download url exposed on the GUI when users right click on a folder and select get url? If yes, I get it why the folder url mapping is needed here as we want to expose uniform url download pattern to users. Otherwise, I think internal folder download url is sufficient.

Reply:
correct, this link is exposed to the user when they select get url  That is why @dtarb would like it to be an external url.